### PR TITLE
Source implementation

### PR DESCRIPTION
**Pull request by voscob**
### Description

This pull request contains the implementation of the Source.

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-cosmos-nosql/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		t.Errorf("unexpected error: %s", err.Error())
+
+		return
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("got: %v, want: %v", got, want)
+	}
+}
```
Comment (maksenius) [inline config/config_test.go:57]: What about negative case?

File: source/config_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		t.Errorf("unexpected error: %s", err.Error())
+
+		return
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("got: %v, want: %v", got, want)
+	}
+}
```
Comment (maksenius) [inline source/config_test.go:51]: What about negative case?

File: source/iterator.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+		}
+
+		// skip the snapshot, so set the most recent item in the container as the last processed item
+		iter.position.LastProcessedValue = latestSnapshotValue
+	}
+
+	return iter, nil
+}
+
+func (iter *iterator) HasNext(ctx context.Context) (bool, error) {
```
Comment (maksenius) [inline source/iterator.go:110]: please add comment

File: source/iterator.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+			return false, fmt.Errorf("load items: %w", err)
+		}
+
+		return iter.items != nil, nil
+	}
+
+	return false, nil
+}
+
+func (iter *iterator) Next() (sdk.Record, error) {
```
Comment (maksenius) [inline source/iterator.go:141]: please add comment

File: source/iterator.go [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+		keys:         config.Keys,
+		orderingKey:  config.OrderingKey,
+		batchSize:    config.BatchSize,
+	}
+
+	if len(iter.keys) == 0 {
+		iter.keys = []string{iter.orderingKey}
+	}
+
+	cred, err := azcosmos.NewKeyCredential(config.PrimaryKey)
```
Comment (maksenius) [inline source/iterator.go:68]: Could you please explain more what we do here

File: source/iterator.go (hunk 3 above)
Comment (voscob) [reply source/iterator.go:114]: Done 👍

File: source/iterator.go (hunk 4 above)
Comment (voscob) [reply source/iterator.go:146]: Done 👍

File: source/iterator.go (hunk 5 above)
Comment (voscob) [reply source/iterator.go:69]: Done 👍

**Review by maksenius — APPROVED** [review]

**Review by alyakimenko — APPROVED** [review]

**Pull request merged**